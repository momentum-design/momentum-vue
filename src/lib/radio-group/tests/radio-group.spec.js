import { mount } from '@vue/test-utils';
import RadioGroup from '../index.vue';
import Radio from '../../radio/index.vue';
import Label from '../../label/index.vue';
import Vue from 'vue';

Vue.component(Radio.name, Radio);
Vue.component(Label.name, Label);

describe('RadioGroup', () => {
  it('should match snapshot', () => {
    const wrapper = mount(RadioGroup, {
      slots: {
        default: `<md-radio htmlId="test" label="test" />`
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle default true Values', () => {
    const wrapper = mount(RadioGroup, {
      propsData: {
        value: 'me'
      },
      slots: {
        default: `<md-radio htmlId="testRadio1" label="me" value="me" />
<md-radio htmlId="testRadio2" label="you" value="you" />
<md-radio htmlId="testRadio3" label="us" value="us" />
`
      }
    });

    expect(wrapper.findAll(Radio).at(0).props().checked).toEqual(true);
    expect(wrapper.findAll(Radio).at(1).props().checked).toEqual(false);
  });

  it('should handle Change Events', () => {
    const wrapper = mount(RadioGroup, {
      slots: {
        default: `<md-radio htmlId="testRadio1" label="me" value="me" />
<md-radio htmlId="testRadio2" label="you" value="you" />
<md-radio htmlId="testRadio3" label="us" value="us" />
`
      }
    });

    wrapper
      .find('#testRadio1')
      .trigger('change');

    expect(wrapper.findAll(Radio).at(0).props().checked).toEqual(true);
    expect(wrapper.findAll(Radio).at(1).props().checked).toEqual(false);
  });
});

