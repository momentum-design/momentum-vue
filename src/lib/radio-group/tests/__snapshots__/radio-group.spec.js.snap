// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should match snapshot 1`] = `
<div class="md-radio-group">
  <div class="md-input-container md-radio"><input aria-checked="true" type="radio" id="test" name="" tabindex="0" class="md-input md-radio__input" value=""> <label for="test" class="md-radio__label"><span>test</span></label> </div>
</div>
`;
